src/Libraries/Microsoft.Extensions.AI/ChatCompletion/FunctionInvokingChatClient.cs

```diff
@@ -1418,7 +1418,11 @@ private static (List<ApprovalResultWithRequestMessage>? approvals, List<Approval
     /// <returns>The <see cref="AIContent"/> for the rejected function calls.</returns>
     private static List<AIContent>? GenerateRejectedFunctionResults(List<ApprovalResultWithRequestMessage>? rejections) =>
         rejections is { Count: > 0 } ?
-            rejections.ConvertAll(static m => (AIContent)new FunctionResultContent(m.Response.FunctionCall.CallId, "Error: Tool call invocation was rejected by user.")) :
+            rejections.ConvertAll(m =>
+            {
+                string result = m.Response.Reason ?? "Error: Tool call invocation was rejected by user.";
```

Comment:
The reason may not include anything that indicates the result shouldn't be considered the actual function result. Consider instead something like:
```suggestion
                string result = "Tool call invocation rejected.";
                if (!string.IsNullOrWhitespace(m.Response.Reason))
                {
                    result = $"{result} {m.Response.Reason}";
                }
```